vllm/v1/worker/gpu_model_runner.py

```diff
@@ -639,10 +639,10 @@ def execute_model(
             assert seq_len <= req_state.num_tokens
             if seq_len == req_state.num_tokens:
                 # Append the sampled token to the output token ids.
-                token_id = sampled_token_ids[i]
-                self.input_batch.token_ids_cpu[i, seq_len] = token_id
                 self.input_batch.num_tokens[i] += 1
-                req_state.output_token_ids.append(token_id)
+                # OPTIMIZATION: Priming the state updates for later updates.
+                req_state.output_token_ids.append(0)
```

Comment:
QQ: Why do we need the placeholder here?

Reply:
Presumably it's to avoid the CPU cycles from list re-sizing/allocation?